lib/cocoapods/installer/analyzer.rb

```diff
@@ -660,18 +660,19 @@ def compute_pod_target_dependencies(pod_targets, all_specs)
         end
 
         pod_targets.each do |target|
-          dependencies = dependencies_for_specs(target.library_specs.to_set, target.platform, all_specs).group_by(&:root)
-          target.dependent_targets = filter_dependencies(dependencies, pod_targets_by_name, target)
-          target.test_dependent_targets_by_spec_name = target.test_specs.each_with_object({}) do |test_spec, hash|
-            test_dependencies = dependencies_for_specs([test_spec], target.platform, all_specs).group_by(&:root)
-            test_dependencies.delete_if { |k, _| dependencies.key? k }
-            hash[test_spec.name] = filter_dependencies(test_dependencies, pod_targets_by_name, target)
+          dependencies_by_config = dependencies_for_specs(target.library_specs.to_set, target.platform, all_specs)
+          target.dependent_targets_by_config = dependencies_by_config.transform_values { |deps| filter_dependencies(deps, pod_targets_by_name, target) }
+
+          target.test_dependent_targets_by_spec_name_by_config = target.test_specs.each_with_object({}) do |test_spec, hash|
+            test_dependencies_by_config = dependencies_for_specs([test_spec], target.platform, all_specs)
+            test_dependencies_by_config.each { |config, deps| deps.delete_if { |k, _| dependencies_by_config[config].key? k } }
+            hash[test_spec.name] = test_dependencies_by_config.transform_values { |deps| filter_dependencies(deps, pod_targets_by_name, target) }
           end
 
-          target.app_dependent_targets_by_spec_name = target.app_specs.each_with_object({}) do |app_spec, hash|
-            app_dependencies = dependencies_for_specs([app_spec], target.platform, all_specs).group_by(&:root)
-            app_dependencies.delete_if { |k, _| dependencies.key? k }
-            hash[app_spec.name] = filter_dependencies(app_dependencies, pod_targets_by_name, target)
+          target.app_dependent_targets_by_spec_name_by_config = target.app_specs.each_with_object({}) do |app_spec, hash|
+            app_dependencies_by_config = dependencies_for_specs([app_spec], target.platform, all_specs)
```

Comment:
@segiddins it seems further up on line 663 we are invoking `dependencies_for_specs` and pass a `Set` versus here we are passing an array.

Reply:
talked offline, can use `Array` here.